keras_nlp/layers/token_and_position_embedding_test.py

```diff
@@ -0,0 +1,141 @@
+        )
+
+        config = token_and_position_embed.get_config()
+
+        expected_config_subset = {
+            "embeddings_initializer": keras.initializers.serialize(
+                keras.initializers.GlorotUniform()
+            ),
+            "position_embeddings_initializer": keras.initializers.serialize(
+                keras.initializers.GlorotUniform()
+            ),
+            "embeddings_regularizer": None,
+            "mask_zero": False,
+        }
+
+        self.assertEqual(config, {**config, **expected_config_subset})
+
+        restored_token_and_position_embed = (
+            TokenAndPositionEmbedding.from_config(config)
+        )
+
+        self.assertEqual(
+            restored_token_and_position_embed.get_config(),
+            {**config, **expected_config_subset},
+        )
+
+    def test_ragged_tensor_with_3_dimensions(self):
+        vocabulary_size = 5
+        sequence_length = 4
+        feature_size = 2
+        embedding_dim = 3
+        test_layer = TokenAndPositionEmbedding(
+            vocabulary_size=vocabulary_size,
+            max_length=sequence_length,
+            embedding_dim=embedding_dim,
+            embeddings_initializer=custom_token_embed_init,
+            position_embeddings_initializer=custom_position_embed_init,
+        )
+        # Create a 3-dimensional ragged input (the first dimension is implicit).
+        input_tensor = tf.keras.Input(
+            shape=(None, feature_size), dtype=tf.float32, ragged=True
+        )
+        output_tensor = test_layer(input_tensor)
+        model = tf.keras.Model(input_tensor, output_tensor)
+
+        input_data = tf.ragged.constant(
+            [
+                [[1.0, 1.0], [1.0, 1.0]],
+                [],
+                [[1.0, 1.0], [1.0, 1.0], [1.0, 1.0]],
+                [[1.0, 1.0]],
+            ],
+            ragged_rank=1,
+            inner_shape=(2,),
+        )
+        expected_output_data = tf.ragged.constant(
+            [
+                [
+                    [[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]],
+                    [[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]],
+                ],
+                [],
+                [
+                    [[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]],
+                    [[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]],
+                    [[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]],
+                ],
+                [[[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]]],
+            ],
+            ragged_rank=1,
+        )
+        output_data = model.predict(input_data)
+        self.assertAllClose(output_data, expected_output_data)
+
+    def test_save_model(self):
+        vocabulary_size = 5
+        sequence_length = 4
+        feature_size = 2
+        embedding_dim = 3
+        test_layer = TokenAndPositionEmbedding(
+            vocabulary_size=vocabulary_size,
+            max_length=sequence_length,
+            embedding_dim=embedding_dim,
+        )
+        inputs = keras.Input(shape=(sequence_length, feature_size))
```

Comment:
The input here should be shape (batch_size, sequence_length). So `keras.Input(shape=(sequence_length,))` here.